src/pam_hbac_obj.c

```diff
@@ -137,8 +144,33 @@ get_user_groups(const char *name, gid_t primary_gid,
         ret = 0;
         *ngroups_ptr = ngroups;
     }
-#else
-#error No known get-groups-for-user implementation found
+#else     // for systems lacking the above functions, tested on hpux only
```

Comment:
Using enumeration is an OK workaround and sudo uses it as well. I would just prefer it it was wrapped in a function of its own, called maybe something like `ph_getgrgid_fallback` ?